app/jobs/slack_event/message_job.rb

```diff
@@ -1,37 +1,108 @@
 #
 # See https://api.slack.com/events-api#receiving_events
 #
-# Example payload:
-#   {"token"=>"xxxxx",
-#   "team_id"=>"T05052K3Q",
-#   "api_app_id"=>"A9MR2EQDQ",
-#   "event"=>{
-#     "type"=>"message",
-#     "user"=>"U6D5W3NCC",
-#     "text"=>"something",
-#     "ts"=>"1526446776.000167",
-#     "channel"=>"C05052K88",
-#     "event_ts"=>"1526446776.000167",
-#     "channel_type"=>"channel"},
-#   "type"=>"event_callback",
-#   "event_id"=>"EvAQE3M1PD",
-#   "event_time"=>1526446776,
-#   "authed_users"=>["U054D8V3L"]}
-#
 class SlackEvent::MessageJob < ApplicationJob
 
   attr_reader :options
   private :options
 
   def perform(options = {})
     @options = options.with_indifferent_access
+
+    %i[
+    request_to_invite
     update_last_message_at
+    ].each do |method|
+      return if send(method)
```

Comment:
this construct reads a little weird to me. is this list likely to change? if not, i'd probably rather see ```
request_to_invite or update_last_message_at
```

Reply:
I suspect it _might_ change. If only because doing something in response to a general message is likely to be quite popular.  I think the better thing to do would be to break this file apart and delegate to "sub jobs" of their own class, but I got tired and didn't :)